lib/ansible/modules/cloud/azure/azure_rm_resource_facts.py

```diff
@@ -183,18 +183,28 @@ def exec_module(self, **kwargs):
 
         header_parameters = {}
         header_parameters['Content-Type'] = 'application/json; charset=utf-8'
-
-        response = self.mgmt_client.query(self.url, "GET", query_parameters, header_parameters, None, [200, 404], 0, 0)
-
-        try:
-            response = json.loads(response.text)
-            if response is list:
-                self.results['response'] = response
-            else:
-                self.results['response'] = [response]
-        except Exception:
-            self.results['response'] = []
-
+        skiptoken = None
+
+        while True:
+            if skiptoken:
+                query_parameters['skiptoken'] = skiptoken
+            response = self.mgmt_client.query(self.url, "GET", query_parameters, header_parameters, None, [200, 404], 0, 0)
+
+            try:
+                response = json.loads(response.text)
+
+                if isinstance(response, list):
```

Comment:
didn't get this, why not check while response.nextPageToken == None?

Reply:
it's not required, it can never be a list